control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1615,56 +1639,68 @@ func setStatusConditionWithTransitionUpdate(conditions *[]metav1.Condition, newC
 	existingCondition.ObservedGeneration = newCondition.ObservedGeneration
 }
 
-func (r *reconciler) ensureCloudResourcesDestroyed(ctx context.Context, hcp *hyperv1.HostedControlPlane) (sets.String, error) {
+func (r *reconciler) ensureCloudResourcesDestroyed(ctx context.Context, hcp *hyperv1.HostedControlPlane) (map[string]bool, error) {
 	log := ctrl.LoggerFrom(ctx)
-	remaining := sets.NewString()
+	remainingIgnoredResources := make(map[string]bool)
 	log.Info("Ensuring resource creation is blocked in cluster")
 	if err := r.ensureResourceCreationIsBlocked(ctx); err != nil {
-		return remaining, err
+		return remainingIgnoredResources, err
 	}
+
 	var errs []error
 	log.Info("Ensuring image registry storage is removed")
-	removed, err := r.ensureImageRegistryStorageRemoved(ctx)
+	removed, ignored, err := r.ensureImageRegistryStorageRemoved(ctx)
```

Comment:
I can't see how the changes to resource.go are adding any value over the existing implementation?
a resource either is removed or not and we are tracking that already, any reason why we can't keep this file as it is?

Reply:
because those dangling objects like PVs/PVCs/LoadBalancers etc... blocks the cluster deprovision.